components/cam/src/control/cam_restart.F90

```diff
@@ -419,31 +417,6 @@ subroutine cam_read_restart( cam_in, cam_out, dyn_in, dyn_out, pbuf2d, &
 
    call phys_grid_init
 
-   
-   if (fv_nphys>0) then
-    ! For FV physgrid we need allocate the phys_state before reading restart data
-    call physics_type_alloc(phys_state, phys_tend, begchunk, endchunk, pcols)
-
-    ! Read topo data from file when physics is on FV grid
-    call cam_initfiles_open()
-    fh_topo => topo_file_get_id()
-    nphys_sq = fv_nphys*fv_nphys
-    allocate(phis_tmp(nphys_sq,nelemd))
-    call infld('PHIS', fh_topo, 'ncol', 1, nphys_sq, 1, nelemd, phis_tmp, found, gridname='physgrid_d')
```

Comment:
Is it ok not reading in the topo data upon the restart? or did you make the model to read PHIS from the restart file?

Reply:
This can get confusing, so let me know if this doesn't make sense. The topo field gets saved to the restart files through the surface geopotential field zs on the dynamics grid. I put this here while I was experimenting with other ways to handle the topo. Currently, the topo is read on the physics grid, mapped to the dynamics grid, and then mapped back to the physics grid, which is pretty weird and introduces extra smoothing. The dyn grid topo is read and mapped to the phys grid on the restart. There was a point where Andrew and I were thinking it might be better to avoid that mapping back and forth and this code was part of that, but it turns out that the extra smoothing from this mapping back and forth seems to be somewhat beneficial. This suggests we need to revise our method of preparing the topo data on the physgrid, but we decided that there are other priorities. I'm planning to come back to this later, but currently this code block is not needed.

Reply:
Got it.